zaza/openstack/charm_tests/cinder_backup/tests.py

```diff
@@ -0,0 +1,182 @@
+    def services(self):
+        """Return a list services for Openstack Release."""
+        current_release = openstack_utils.get_os_release()
+        services = ['cinder-scheduler', 'cinder-volume']
+        if (current_release >=
+                openstack_utils.get_os_release('xenial_ocata')):
+            services.append('apache2')
+        else:
+            services.append('cinder-api')
+        return services
+
+    def test_100_volume_create_extend_delete(self):
+        """Test creating, extending a volume."""
+        vol_new = openstack_utils.create_volume(
+            self.cinder_client,
+            name='{}-100-vol'.format(self.RESOURCE_PREFIX),
+            size=1)
+        self.cinder_client.volumes.extend(
+            vol_new.id,
+            '2')
+        openstack_utils.resource_reaches_status(
+            self.cinder_client.volumes,
+            vol_new.id,
+            expected_status="available",
+            msg="Volume status wait")
+
+    def test_410_cinder_vol_create_backup_delete_restore_pool_inspect(self):
+        """Create, backup, delete, restore a ceph-backed cinder volume.
+
+        Create, backup, delete, restore a ceph-backed cinder volume, and
+        inspect ceph cinder pool object count as the volume is created
+        and deleted.
+        """
+        unit_name = 'ceph-mon/0'
+        obj_count_samples = []
+        pool_size_samples = []
+        pools = ceph_utils.get_ceph_pools(unit_name)
+        expected_pool = 'cinder-ceph'
+        cinder_ceph_pool = pools[expected_pool]
+
+        # Check ceph cinder pool object count, disk space usage and pool name
+        logging.info('Checking ceph cinder pool original samples...')
+        pool_name, obj_count, kb_used = ceph_utils.get_ceph_pool_sample(
+            unit_name, cinder_ceph_pool)
+
+        obj_count_samples.append(obj_count)
+        pool_size_samples.append(kb_used)
+
+        self.assertEqual(pool_name, expected_pool)
+
+        # Create ceph-backed cinder volume
+        cinder_vol = openstack_utils.create_volume(
+            self.cinder_client,
+            name='{}-410-vol'.format(self.RESOURCE_PREFIX),
+            size=1)
+        # Backup the volume
+        vol_backup = openstack_utils.create_volume_backup(
+            self.cinder_client,
+            cinder_vol.id,
+            name='{}-410-backup-vol'.format(self.RESOURCE_PREFIX))
+        # Delete the volume
+        openstack_utils.delete_volume(self.cinder_client, cinder_vol.id)
+        # Restore the volume
+        self.cinder_client.restores.restore(vol_backup.id)
+        openstack_utils.resource_reaches_status(
+            self.cinder_client.backups,
+            vol_backup.id,
+            expected_status='available',
+            msg='Backup status wait')
+        # Delete the backup
+        openstack_utils.delete_volume_backup(self.cinder_client, vol_backup.id)
+
+        # Re-check ceph cinder pool object count and disk usage
+        time.sleep(10)
```

Comment:
Rather than sleeping, can you use resource_removed to wait for the resource to be removed ? There's an example of it being used in the [cinder tests](https://github.com/openstack-charmers/zaza-openstack-tests/blob/master/zaza/openstack/charm_tests/cinder/tests.py#L135)